app/controllers/videos_controller.rb

```diff
@@ -0,0 +1,15 @@
+class VideosController < ApplicationController
+  before_action :require_user
+  def index
+    @videos = Video.all
+    @categories = Category.all
+  end
+  def show
+    @video = Video.find(params[:id])
+  end
+
+  def search
+    @videos = Video.search_by_title(params[:title])
+    render 'search'
```

Comment:
Why are you using `render 'search'` ? It is standard Rails behavior to render a template of the same name as the action.